File: Exercism/c/rna-transcription/src/rna_transcription.c
#include <stdlib.h>
#include <string.h>
#include "rna_transcription.h"

char* to_rna(const char* dna) {
    char* rna = malloc(strlen(dna));

    for (size_t i = 0 ; i < strlen(dna) ; ++i) {
        switch (dna[i]) {
        case 'G':
            rna[i] = 'C';
            break;

        case 'C':
            rna[i] = 'G';
            break;

        case 'T':
            rna[i] = 'A';
            break;

        case 'A':
            rna[i] = 'U';
            break;

        default:
            free(rna);
            return NULL;
        }
    }

    return rna;
}

File: Exercism/fortran/rna-transcription/rna_transcription.f90
module rna_transcription
   implicit none
contains
   function translate(c) result(r)
      character :: c, r
      select case (c)
      case ('C')
         r = 'G'
      case ('G')
         r = 'C'
      case ('T')
         r = 'A'
      case ('A')
         r = 'U'
      end select
   end function translate

   function to_rna(dna)
      integer :: i
      character(*) :: dna
      character(len(dna)) :: to_rna
      do i = 1, len(dna)
         to_rna(i:i) = translate(dna(i:i))
      end do
   end function to_rna

end module rna_transcription
